lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -985,6 +986,56 @@ void BoundInterpreterFunction::fwdCrossEntropyLossGradInst(
   }
 }
 
+//===----------------------------------------------------------------------===//
+//                       Peer-to-peer Communication
+//===----------------------------------------------------------------------===//
+
+void BoundInterpreterFunction::fwdSendInst(const SendInst *I) {
+  using TensorPipeTy = std::pair<std::future<Tensor>, std::promise<Tensor>>;
```

Comment:
I would prefer to avoid this. If I move the definitions of these types to `Interpreter.h`, they'll be available here but not in the DeviceManager which also needs them.